R/pkg/R/deserialize.R

```diff
@@ -174,11 +174,46 @@ readRow <- function(inputCon) {
 }
 
 # Take a single column as Array[Byte] and deserialize it into an atomic vector
-readCol <- function(inputCon, numRows) {
-  # sapply can not work with POSIXlt
-  do.call(c, lapply(1:numRows, function(x) {
-    value <- readObject(inputCon)
-    # Replace NULL with NA so we can coerce to vectors
-    if (is.null(value)) NA else value
-  }))
+readCol <- function(inputCon, numRows, colType) {
+  if (numRows > 0) {
+    # sapply can not work with POSIXlt
+    do.call(c, lapply(1:numRows, function(x) {
+      value <- readObject(inputCon)
+      # Replace NULL with NA so we can coerce to vectors
+      if (is.null(value)) NA else value
+    }))
+  } else {
+    vector(mode = colType)
```

Comment:
So fundamentally it looks like this we want to create a vector with `mode = colType` if the column is empty. But does it matter if we get the `colType` wrong ? i.e. what happens if we just use the default mode (which I think is `logical`)